### PR TITLE
registration email feature implemented

### PR DESCRIPTION
**Pull request by bvishal8510**
@apoorvaeternity please review the changes along with design changes in blog, blog details, index and registration page.

### CODE REVIEW

File: Core/models.py [hunk 1]
```diff
@@ -217,3 +218,22 @@ def get_absolute_url(self):
     def __str__(self):
         return "{} | {}".format(self.title, self.author)
 
+
+class Email_Content(models.Model):
```
Comment (animesh21) [inline Core/models.py]: class name should be in UpperCamelCase not Upper_Camel_Case_Mixed_With_Snake_Case
```suggestion
class EmailContent(models.Model):
```

File: Core/models.py [hunk 2]
```diff
@@ -217,3 +218,22 @@ def get_absolute_url(self):
...
+    event = models.ForeignKey('Event')
+    mail_allowed = models.BooleanField(default=False)
+    subject = models.CharField(max_length=100)
+    text = RichTextField()
+
+    def __str__(self):
+        return self.event.name
+
+
+class Email_Attachment(models.Model):
```
Comment (animesh21) [inline Core/models.py]: .

File: Core/views.py [hunk 3]
```diff
@@ -2,13 +2,16 @@
...
-from django.core.mail import send_mail
 from django.core.urlresolvers import reverse_lazy
 from django.template.loader import render_to_string
 
 from .forms import ContactUsForm, RegistrationForm
-from .models import Project, Member, ContactInfo, Blog, Event, ContactUs, Registration
+from .models import Project, Member, ContactInfo, Blog, Event, ContactUs, Registration, \
+ Email_Content, Email_Attachment
 from IceCream.settings import RECEIVER_EMAIL, EMAIL_HOST_USER
+from django.core.mail import send_mail, EmailMessage
```
Comment (animesh21) [inline Core/views.py:7]: Arrange imports as per [PEP8](https://www.python.org/dev/peps/pep-0008/#imports)

File: Core/views.py [hunk 4]
```diff
@@ -116,8 +119,44 @@ class RegistrationView(FormView):
...
+
         if form.is_valid():
+            event_receiver_email = form.cleaned_data['email']
             form.save()
+
+            allowed = False
+
+            try:
+                allowed = (Email_Content.objects.get(event=self.event)).mail_allowed
+            except:
```
Comment (animesh21) [inline Core/views.py]: Please don't use bare `except` clauses, catch specific errors like
```suggestion
            except Email_Content.DoesNotExist:
```

File: Core/views.py [hunk 5]
```diff
@@ -116,8 +119,44 @@ class RegistrationView(FormView):
...
+                })
+
+                from_mail = EMAIL_HOST_USER
+                to_mail = [event_receiver_email]
+
+                mail = EmailMessage(subject, message, from_mail, to_mail)
+                mail.content_subtype = "html"
+                mail.mixed_subtype = 'related'
+
+                image_sub_type = ((str(self.event.pic_path)).split('.'))[-1]
```
Comment (animesh21) [inline Core/views.py]: remove redundant parenthesis
```suggestion
                image_sub_type = str(self.event.pic_path).split('.')[-1]
```

File: IceCream/settings.py [hunk 6]
```diff
@@ -22,7 +22,7 @@
 SECRET_KEY = 'et5csce+0jh4-8!u2s#5n!d!6nvnrs*0uw^)mgfd+(@m1wr&oz'
 
 # SECURITY WARNING: don't run with debug turned on in production!
-# DEBUG = False
+# DEBUG = True
```
Comment (animesh21) [inline IceCream/settings.py]: These commented lines should be removed and the value of `DEBUG` should be taken from environment variable instead.

**Review by animesh21 — CHANGES_REQUESTED** [review]
Please take care of the points I've mentioned.

**Review by parammittal16 — COMMENTED** [review]
```
@bvishal8510 there are some links in footer pointing to https://akgec.in/ 
which is now https://akgec.ac.in/
so the URL to all those links can be updated !!
```

File: Core/models.py [hunk 7]
```diff
@@ -217,3 +218,22 @@ def get_absolute_url(self):
...
+
+
+class EmailAttachment(models.Model):
+    event = models.ForeignKey('Event')
+    files = models.FileField(upload_to=events_upload_location, null=True,
+                                default=None)
+    name = models.CharField(max_length=80, blank=False)
+
+    def __str__(self):
+        return self.event.name + "  " + self.name
```
Comment (apoorvaeternity) [inline Core/models.py]: PEP8!!! No new line at the end of file.

File: Core/templates/blog.html [hunk 8]
```diff
@@ -191,31 +188,31 @@ <h5 class="font-weight-bold text-uppercase mt-3 mb-4">Clients</h5>
                     
                     <ul class="list-unstyled">
                         <li>
-                            <a href="http://akgec.in/">AKGEC</a>
+                            <a target="_blank" href="https://akgec.ac.in/">AKGEC</a>
```
Comment (apoorvaeternity) [inline Core/templates/blog.html:191]: Try to keep unrelated changes in separate pull requests. For eg. this PR is related to registration email feature.

**Review by apoorvaeternity — CHANGES_REQUESTED** [review]

File: IceCream/settings.py (hunk 6 above)
Comment (apoorvaeternity) [reply IceCream/settings.py]: @bvishal8510 Or else you can also create separate development and production settings(for eg. create a directory settings with files base.py with base settings and development.py for development settings and production.py for production settings) and set the environment variable `DJANGO_SETTINGS_MODULE` to specify the settings to be used.

**Comment by apoorvaeternity** [comment]
@bvishal8510 Please create a separate issue for every change that is required on the website. **It helps in better tracking of why a particular change was needed**. Refer those issue from the PR so that they can be closed once the issue is fixed.

**Review by animesh21 — APPROVED** [review]
